src/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Hll.cpp

```diff
@@ -0,0 +1,294 @@
+    const gsl::not_null<Scalar<DataVector>*>
+        packaged_largest_ingoing_char_speed,
+
+    const Scalar<DataVector>& mass_density,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& momentum_density,
+    const Scalar<DataVector>& energy_density,
+
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_mass_density,
+    const tnsr::IJ<DataVector, Dim, Frame::Inertial>& flux_momentum_density,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_energy_density,
+
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& velocity,
+    const Scalar<DataVector>& specific_internal_energy,
+
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+    /*mesh_velocity*/,
+    const std::optional<Scalar<DataVector>>& normal_dot_mesh_velocity,
+    const EquationsOfState::EquationOfState<false, ThermodynamicDim>&
+        equation_of_state) const noexcept {
+  {
+    // Compute sound speed
+    Scalar<DataVector>& sound_speed = *packaged_mass_density;
+    Scalar<DataVector>& normal_dot_velocity = *packaged_energy_density;
+    sound_speed_squared(make_not_null(&sound_speed), mass_density,
+                        specific_internal_energy, equation_of_state);
+    get(sound_speed) = sqrt(get(sound_speed));
+    dot_product(make_not_null(&normal_dot_velocity), velocity, normal_covector);
+    // Compute the largest outgoing / ingoing characteristic speeds. Note that
+    // the notion of being 'largest ingoing' means taking the most negative
+    // value given that the positive direction is outward normal.
+    if (normal_dot_mesh_velocity.has_value()) {
+      get(*packaged_largest_outgoing_char_speed) =
+          get(normal_dot_velocity) + get(sound_speed) -
+          get(*normal_dot_mesh_velocity);
+      get(*packaged_largest_ingoing_char_speed) =
+          get(normal_dot_velocity) - get(sound_speed) -
+          get(*normal_dot_mesh_velocity);
+    } else {
+      get(*packaged_largest_outgoing_char_speed) =
+          get(normal_dot_velocity) + get(sound_speed);
+      get(*packaged_largest_ingoing_char_speed) =
+          get(normal_dot_velocity) - get(sound_speed);
+    }
+  }
+
+  *packaged_mass_density = mass_density;
+  *packaged_momentum_density = momentum_density;
+  *packaged_energy_density = energy_density;
+
+  dot_product(packaged_normal_dot_flux_mass_density, flux_mass_density,
+              normal_covector);
+  for (size_t i = 0; i < Dim; ++i) {
+    packaged_normal_dot_flux_momentum_density->get(i) =
+        get<0>(normal_covector) * flux_momentum_density.get(0, i);
+    for (size_t j = 1; j < Dim; ++j) {
+      packaged_normal_dot_flux_momentum_density->get(i) +=
+          normal_covector.get(j) * flux_momentum_density.get(j, i);
+    }
+  }
+  dot_product(packaged_normal_dot_flux_energy_density, flux_energy_density,
+              normal_covector);
+
+  // What value to return here..?
```

Comment:
I left this comment to ask if `dg_package_data` should return a value for HLL case -- currently the return type is void. (For Rusanov implementation, max value of `packaged_abs_char_speed` is returned here)

Reply:
You have to use `double` as the return type. I'm preparing a PR that will make this fail tests. Every boundary correction needs to have the same interface, which means returning a `double`. The value of the double should be an estimate of the max characteristic speed on the interface

Reply:
By the max characteristic speed do you mean the one with maximum absolute value (whether it is ingoing or outgoing)?